services/web/client/source/class/qxapp/data/Permissions.js

```diff
@@ -76,6 +83,56 @@ qx.Class.define("qxapp.data.Permissions", {
   members: {
     __userRole: null,
 
+    addAction: function(role, action) {
+      if (!this.self().ROLES[role]) {
+        return;
+      }
+
+      this.self().ACTIONS[action] = this.__nextAction();
+      this.self().ROLES[role].can.push(action);
+    },
+
+    canDo: function(action, showMsg) {
+      let canDo = false;
+      if (this.__userRole) {
+        canDo = this.__canRoleDo(this.__userRole, action);
+      }
+      if (showMsg && !canDo) {
+        qxapp.component.widget.FlashMessenger.getInstance().logAs("Operation not permitted", "ERROR");
+      }
+      return canDo;
+    },
+
+    loadUserRoleFromBackend: function() {
+      let userResources = qxapp.io.rest.ResourceFactory.getInstance().createUserResources();
+
+      let profile = userResources.profile;
+      profile.addListenerOnce("getSuccess", e => {
+        let profileData = e.getRequest().getResponse().data;
+        this.__userRole = profileData.role;
+        this.fireDataEvent("userProfileRecieved", true);
+      }, this);
+      profile.addListenerOnce("getError", e => {
+        console.error(e);
+      });
+      profile.get();
+    },
+
+    __getInitPermissions: function() {
```

Comment:
I like the design choice of creating a map between a role and its associated actions.

Nonetheless, i would encourage you to design some hierarchical structure to define actions instead of adding random strings as actions, i.e. some kind of combinations that with a few switches allow describing a wide range of actions. Current approach definitively works but **is not maintainable** at long term.

I also find a big issue with the fact that the server does not know anything about this. The UI, might not allow access to some of the UI components but if I can still use the server API, then it is not really protecting the action (e.g. can always execute it from a different client). IMHO the permission policy should be defined from the server up to the client and not the other way around.

Reply:
yes ... this can not be stressed enough!

Reply:
Agree, closing PR